iwyu.cc

```diff
@@ -3596,6 +3596,11 @@ class IwyuAstConsumer
     ParseFunctionTemplates(context.getTranslationUnitDecl());
 
     TraverseDecl(context.getTranslationUnitDecl());
+
+    // Check if any unrecoverable errors have occurred
+    // For ex/ No sense continuing if an #include-ed file isn't found
+    if (compiler()->getDiagnostics().hasUnrecoverableErrorOccurred())
+      exit(EXIT_INVALIDARGS);
```

Comment:
I'm not so sure about the `EXIT_INVALIDARGS` here, that seems to be used to signal invalid arguments to IWYU.

But apparently we never reserved room for more failure exit codes :-(, so I think this is the best we can do.

Reply:
Noted in latest change that this a least-bad option and todo-ed when the exit codes get reworked.